### PR TITLE
[VPU] WA for Segmentation fault on dlclose() issue

### PR DESCRIPTION
**Pull request by AndrewBakalinIntel**
Ticket - #-49444
Prevent myriadPlugin to be unloaded at runtime.

### CODE REVIEW

**Comment by AndrewBakalinIntel** [comment]
@ilya-lavrenov could you please take a look? Is it ok to apply such a WA? And how the risk can be evaluated?

**Comment by ilya-lavrenov** [comment]
Do we know why only MYRIAD plugin is affected? Which global resources it has?

**Comment by AndrewBakalinIntel** [comment]
@ilya-lavrenov we have actually the same scenario but on MacOS and with another client application (#-26514). There are already some problems specifically for MyriadPlugin which you may find in the discussions of this ticket. So we can definitely say that MyriadPlugin library is not safe to be `dlclose()`d (or more precisely, unloaded from the memory at runtime) by its design.

**Comment by ilya-lavrenov** [comment]
All plugins are unloaded in runtime once the IE library is unloaded (e.g. in our sample), but IE is loaded as usual library during application start. And I don't understand the different when IE library is unloaded using `dlclose` as well.

**Comment by AndrewBakalinIntel** [comment]
@ilya-lavrenov It seems like we always had this problem, but in most of the cases it doesn't lead to segfault for some unknown reasons. I was able to reproduce this segfault only using GCC 9.3.0, LTO enabled and in Release mode. Even in this conditions client application sometimes exits without segfault.

File: inference-engine/src/vpu/myriad_plugin/CMakeLists.txt [hunk 1]
```diff
@@ -37,9 +37,15 @@ endif()
 # sem_* POSIX functions must be overriden on OSX,
 # these functions come from "mvnc" target,
 # "mvnc" must be the first library in the link list
+set(LINK_LIBRARIES)
+list(APPEND LINK_LIBRARIES mvnc inference_engine inference_engine_legacy vpu_graph_transformer)
+# MyriadPlugin is not safe to be unloaded from memory at runtime
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    list(APPEND LINK_LIBRARIES "-Wl,-z,nodelete")
```
Comment (ilya-lavrenov) [inline inference-engine/src/vpu/myriad_plugin/CMakeLists.txt:44]: it's not a static option I suppose. Can we use approach I suggested in the commend above?

Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/myriad_plugin/CMakeLists.txt]: Done.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by Maxim-Doronin** [comment]
@AndrewBakalinIntel, had myriadx devices already been closed when segfault occurred?

**Comment by AndrewBakalinIntel** [comment]
> @AndrewBakalinIntel,
> 
> * had myriadx devices already been closed when segfault occurred?
> * is `libusb_exit` called with your changes? Is it ok, if no?

@Maxim-Doronin 
1) Yes, `closeDevices` method is called.
2) `libusb_exit` is called in both cases, without my changes - when client application calls `dlclose` on inference engine lib, with my changes - when client application terminates.

**Review by Maxim-Doronin — APPROVED** [review]
> Yes, `closeDevices` method is called.

It turns that this issue is not the same on MacOS as I can see

I have no objections. Can we cherry-pick it into the release branch?
What do you think should we create a new ticket to fix this behavior and remove WA?

**Pull request merged**